src/Microsoft.TestPlatform.PlatformAbstractions/net451/System/ProcessHelper.cs

```diff
@@ -15,21 +17,93 @@ namespace Microsoft.VisualStudio.TestPlatform.PlatformAbstractions;
 
 public partial class ProcessHelper : IProcessHelper
 {
+    [DllImport("kernel32.dll", SetLastError = true)]
+    private static extern bool IsWow64Process2(IntPtr process, out ushort processMachine, out ushort nativeMachine);
+
+    private const ushort Arm64Machine = 0xAA64;
+
     /// <inheritdoc/>
     public string GetCurrentProcessLocation()
-    {
-        return Path.GetDirectoryName(GetCurrentProcessFileName());
-    }
+        => Path.GetDirectoryName(GetCurrentProcessFileName());
 
+    /// <inheritdoc/>
     public IntPtr GetProcessHandle(int processId)
+        => Process.GetProcessById(processId).Handle;
+
+    /// <inheritdoc/>
+    public PlatformArchitecture GetCurrentProcessArchitecture() =>
+        IntPtr.Size == 8 ?
+        IsArm64() ? PlatformArchitecture.ARM64 : PlatformArchitecture.X64
+        : PlatformArchitecture.X86;
+
+    private static bool IsArm64()
     {
-        return Process.GetProcessById(processId).Handle;
+        try
+        {
+            var currentProcess = Process.GetCurrentProcess();
+            if (!IsWow64Process2(currentProcess.Handle, out ushort processMachine, out ushort nativeMachine))
+                throw new Win32Exception();
+
+            // if IMAGE_FILE_MACHINE_UNKNOWN we're not running inside WOW64 and we can verify the native architecture.
+            if (processMachine == 0 && nativeMachine == Arm64Machine)
+            {
+                // To distinguish between ARM64 and x64 emulated on ARM we check the PE header.
+                return IsArm64Module(currentProcess.MainModule.FileName);
+            }
+        }
+        catch (Exception)
+        {
+            // We should have Win32Exception() or IOException() but we don't want to block and we'll fallback to emulated path.
+        }
+
+        return false;
     }
 
-    /// <inheritdoc/>
-    public PlatformArchitecture GetCurrentProcessArchitecture()
+    private static bool IsArm64Module(string path)
```

Comment:
IsArm64Executable?

Might be less correct, but is easier to understand.

Reply:
will do thx